contracts/finance/andromeda-splitter/src/contract.rs

```diff
@@ -216,6 +252,69 @@ fn execute_send(
         .add_attribute("sender", info.sender.to_string()))
 }
 
+fn execute_send_cw20(
+    ctx: ExecuteContext,
+    amount: Uint128,
+    asset: String,
+    config: Option<Vec<AddressPercent>>,
+) -> Result<Response, ContractError> {
+    let ExecuteContext { deps, info, .. } = ctx;
+    let splitter = SPLITTER.load(deps.storage)?;
+
+    let splitter_recipients = if let Some(config) = config {
+        validate_recipient_list(deps.as_ref(), config.clone())?;
+        config
+    } else {
+        splitter.recipients
+    };
+
+    let mut msgs: Vec<SubMsg> = Vec::new();
+    let mut amp_funds: Vec<Coin> = Vec::new();
+    let mut remainder_funds = coin(amount.u128(), asset.clone());
+
+    for recipient_addr in splitter_recipients {
+        let recipient_percent = recipient_addr.percent;
+        let mut vec_coin: Vec<Coin> = Vec::new();
+        let coin = coin(amount.u128(), asset.clone());
+        let amount_owed = coin.amount.mul_floor(recipient_percent);
+
+        if !amount_owed.is_zero() {
+            let mut recip_coin: Coin = coin.clone();
+            recip_coin.amount = amount_owed;
+            remainder_funds.amount = remainder_funds.amount.checked_sub(recip_coin.amount)?;
+            vec_coin.push(recip_coin.clone());
+            amp_funds.push(recip_coin.clone());
+            let amp_msg = recipient_addr.recipient.generate_msg_cw20(
+                &deps.as_ref(),
+                Cw20Coin {
+                    address: recip_coin.denom.clone(),
+                    amount: recip_coin.amount,
+                },
+            )?;
+            msgs.push(amp_msg);
+        }
+    }
```

Comment:
I think this is a fine temporary solution but we should consider adding CW20 support to Kernel routing in the near future.